### PR TITLE
Main controller refactoring

### PR DESCRIPTION
**Pull request by mauroz77**

### CODE REVIEW

File: src/app/mapping.service.ts [hunk 1]
```diff
@@ -11,13 +11,14 @@ import { SummaryInterface } from './summary-interface';
...
-    private serverUrl = 'http://localhost:8081'; // this.devServer; //
+    private SERVER_URL = 'http://localhost:8081'; // this.devServer; //
+    private BASE_URL = this.SERVER_URL + '/api/mappings';
 
-    private summaryUrl = this.serverUrl + '/api/mappings/summary';
-    private mappingsUrl = this.serverUrl + '/api/mappings';
-    private missingMappingsUrl = this.serverUrl + '/api/getmissingdiagnosismappings';
-    public exportUrl = this.serverUrl + '/api/mappings/export';
-    private uploadURL = this.serverUrl + '/api/mappings/uploads';
+    private SUMMARY_URL = this.BASE_URL + '/summary';
```
Comment (Afollet) [inline src/app/mapping.service.ts:17]: What is the reason for the upper-casing? Just a curiosity.

File: sonar-project.properties [hunk 2]
```diff
@@ -1,11 +1,12 @@
 sonar.host.url=http://localhost:9000 
 sonar.login=admin
-sonar.password=admin
+sonar.password=pdx-admin-ui
```
Comment (Afollet) [inline sonar-project.properties:3]: Can you link to information in the Confluence or in the readme how to run SonarQube locally for the project?

**Review by Afollet — APPROVED** [review]
The tests look great.

File: src/app/mapping.service.ts (hunk 1 above)
Comment (mauroz77) [reply src/app/mapping.service.ts:17]: Just as a convention for constant values. https://google.github.io/styleguide/javascriptguide.xml?showone=Constants#Constants (_Use NAMES_LIKE_THIS for constant values._)

File: sonar-project.properties (hunk 2 above)
Comment (mauroz77) [reply sonar-project.properties:3]: This is the link in confluence: https://www.ebi.ac.uk/seqdb/confluence/display/MouseInformatics/PDX+Admin.
I think putting it in readme is also a good idea but I'll do that in another PR

File: src/app/mapping.service.ts (hunk 1 above)
Comment (Afollet) [reply src/app/mapping.service.ts:17]: Yes, that makes sense. Because of my nativity with Angular I do not understand how these are defined constant.

Comment (Afollet) [reply src/app/mapping.service.ts:17]: For example, why wouldn't "devServer" be a constant?

Comment (mauroz77) [reply src/app/mapping.service.ts:17]: I see your point. That's something that can be improved. We 
could explicitly set them as readonly (I'd say the equivalent to final in Java) https://www.typescriptlang.org/docs/handbook/classes.html#readonly-modifier. I'll see if it works well and update the PR

**Pull request merged**